packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -826,15 +826,15 @@ type Insertable<T extends ImplicitAllowedTypes> = readonly (
 	| IterableTreeArrayContent<InsertableTreeNodeFromImplicitAllowedTypes<T>>
 )[];
 
-abstract class CustomArrayNodeBase<const T extends ImplicitAllowedTypes>
+abstract class CustomArrayNodeBase<const T extends ImplicitAnnotatedAllowedTypes>
```

Comment:
I'd like to see a unit test added to src/test/simple-tree/arrayNode.spec.ts testing the case of moving a node to a location where it requires enablement.

Such moves need to error if the type has not been enabled for its destination yet. This might work as is, but should have a test to ensure it works and stays working.

We should also test such a move after the enablement has happened (another client did a schema upgrade). I'm not actually sure if the intention is to allow such a move or not (clarity on what is allowed in this case is missing from the docs) but either way a test sould be good.

Reply:
> We should also test such a move after the enablement has happened (another client did a schema upgrade). I'm not actually sure if the intention is to allow such a move or not.

This out of my lane, but my two cents is that it would be preferrable to NOT allow such a move since that would complicate the guarantees associated with this feature. No matter what we pick, I would bet that the vast majority of app authors would looks at a client whose view schema has `enablable(Foo) ` to think "this client couldn't have attached any `Foo` instances in this field", so we may as well pick the implementation choice that matches that. I may be missing some strong incentive to go the other way though.

Reply:
Thats fine with me. As long as we have some policy, which is documented and tested, and it ensures we don't corrupt documents with out of schema items getting moved in, I'm happy. Yann's suggestion sounds like a fine choice as long as we test and enforce it.